fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java

```diff
@@ -218,16 +286,26 @@ private CacheEntry newCacheEntry() {
                     if (bytesRead < bytesToRead) {
                         // Read less than it should be, possibly hitting EOF.
                         // Set smaller ByteBuffer limit and break out of the loop.
-                        buffers.add(ByteBuffer.wrap(buf, 0, bytesRead));
+                        positionOutputStream.write(buf, 0, bytesRead);
                         break;
                     } else {
-                        buffers.add(ByteBuffer.wrap(buf));
+                        positionOutputStream.write(buf);
                     }
                 }
-
                 stream.close();
-                return new CacheEntry(fileLength - totalBytesToRead, buffers);
+                positionOutputStream.close();
+                hadoopOutputFile.getFileSystem().rename(
+                        hadoopOutputFile.getPath(),
+                        IOUtil.s3aToLocalFilePath(METADATA_CACHE_DISK_PATH, location()));
```

Comment:
It looks like not work when the meta file is at HDFS

Reply:
yes, need to adapt hdfs.

Reply:
Could you design without relying on a specific file system?  so we don't have to be compatible with various file systems